### PR TITLE
remove extra checked out and compiled files for ocean/ice components

### PR DESCRIPTION
**Pull request by JosephMouallem**
This PR aims to clean up the ocean/ice code used for SHiEMOM by removing extra checked-out and compiled codes.
The test case with the new exec runs Ok.

### CODE REVIEW

File: CHECKOUT_mom6.csh
Comment (lharris4) [inline CHECKOUT_mom6.csh:13]: Long-term I don't think it is a good idea to hard-code individual commit hashes. For this stage of development it is fine and probably necessary.

File: Build/BUILDmom6
Comment (lharris4) [inline Build/BUILDmom6:99]: Does MOM6 not support OpenMP or is there just not a compiler flag for this?

**Review by lharris4 — APPROVED** [review]
Fewer checkouts and a more streamlined code are always good. It will probably be a while before we would need to worry about the BGC (although I can imagine there will eventually be interest in this).

File: Build/BUILDmom6
Comment (JosephMouallem) [reply Build/BUILDmom6:99]: If used in debug mode, it was causing that noise in the ocean fields, which apparently was known issue.
In repro mode, parsing some files was causing a crash and it looks like compiling without openmp is solving it, although I believe it requires further investigation.

Comment (lharris4) [reply Build/BUILDmom6:99]: Thank you for describing this issue. Is this an issue with the coupler or with MOM6?

Comment (JosephMouallem) [reply Build/BUILDmom6:99]: It is from a MOM6 file called MOM_file_parser

Comment (bensonr) [reply Build/BUILDmom6:99]: MOM6 doesn't do OpenMP well at all, so this is a good choice.

**Review by bensonr — APPROVED** [review]

File: CHECKOUT_mom6.csh
Comment (breichl) [inline CHECKOUT_mom6.csh:21]: This gives error "popd: Directory stack empty."

File: CHECKOUT_mom6.csh
Comment (breichl) [reply CHECKOUT_mom6.csh:13]: Is it necessary to use this commit or does it work with dev/gfdl?  Is the motivation for using the static commit that we don't have CI setup?

Comment (JosephMouallem) [reply CHECKOUT_mom6.csh:13]: I haven't tried the latest dev/GFDL version, so I'm not sure if it will work right off the bat. If not, I don't anticipate many changes since this is not an old commit.
The static commit is taken from the original OMIP version we first started working on to get a prototype model up and running. It is probably a good idea to move to the most recent ocean version if you guys are comfortable with it.

Comment (lharris4) [reply CHECKOUT_mom6.csh:13]: @breichl which MOM6 branch/tag/checkout do you recommend for us to use as a reasonably-stable version for coupled-model development? We had a similar question from Steve P at Sofar.

Comment (breichl) [reply CHECKOUT_mom6.csh:13]: One option if we wanted something stable and tested outside of GFDL is to use https://github.com/mom-ocean/MOM6.  This is probably the recommendation I would give to outside groups, it is updated less frequently and is tested by a consortium before merging updates.  But I would not prefer this approach for our development since it will not give us the ability to modify source code efficiently.

We could create our own fork and merge updates from dev/gfdl at semi-regular intervals.  The benefit is that we have control over what gets merged in and we can easily update source code.  But I don't think we need this level of control so this is probably not right for us either.

The more I think about it, what Joseph is doing is consistent with how we develop other models (e.g., OM4, OM5) where we point to a specific hash either in a .gitmodules file or in an XML/script and update the hash when we know there are changes to dev/GFDL that we want.  We can easily push code changes to dev/gfdl to update source code.  So I'm actually happy with this, we just need to remember to periodically update it.  Maybe we set these hashes near the top of the script as "MOM_VERSION" and "SIS_VERSION" to make it clear?

Comment (bensonr) [reply CHECKOUT_mom6.csh:13]: Doesn't the dev/gfdl branch of MOM6 use tags to ensure people know what the stable releases are?  This idea of relying on hashes which comes with no overall description makes it hard to know at a glance where you are in the tree related to the top of the trunk.  At a minimum, is there a changelog or some other plain-language document describing the state of the hash?

Comment (JosephMouallem) [reply CHECKOUT_mom6.csh:13]: @breichl, Initially, I worked on forked branches of each component to get things up and running. Then, we merged all updates into our forks, and after some testing, we pushed our changes to the main repositories.

Now, we are using the latest code updates, as mentioned in the other CHECKOUT_code file. We could continue using the latest dev/gfdl branch to keep receiving all the updates. Once a significant implementation needs to be done and tested, the developer could work on their fork and then merge back (essentially similar to the process I initially described).

Comment (breichl) [reply CHECKOUT_mom6.csh:13]: > Doesn't the dev/gfdl branch of MOM6 use tags to ensure people know what the stable releases are? This idea of relying on hashes which comes with no overall description makes it hard to know at a glance where you are in the tree related to the top of the trunk. At a minimum, is there a changelog or some other plain-language document describing the state of the hash?

It may help to check w/ Alistair/Marshall/Bob to clarify, but here is how I understand it:

NOAA-GFDL/MOM6 hasn't really been traditionally tagged/released, except a few times before 2020 (e.g., for the Adcroft et al 2019 paper).  The "releases" are facilitated through the main branch at the mom-ocean/MOM6 level, since there is now a community of external groups also developing MOM6.  mom-ocean also doesn't use tags, but is only updated every couple of months when a member of the consortium pushes their dev branch.  I think the PR and commit summaries serve as the primary description of the change/release.

At GFDL I have seen date strings also added along with the hash in the XML, maybe that was to make it something a bit more interpretable but I do not know (I think only the hash actually determines the checkout, so the date strings seem sort of arbitrary otherwise).  Otherwise I think we also rely on the commit logs (which are enforced to be descriptive and include all details).



> @breichl, Initially, I worked on forked branches of each component to get things up and running. Then, we merged all updates into our forks, and after some testing, we pushed our changes to the main repositories.
> 
> Now, we are using the latest code updates, as mentioned in the other CHECKOUT_code file. We could continue using the latest dev/gfdl branch to keep receiving all the updates. Once a significant implementation needs to be done and tested, the developer could work on their fork and then merge back (essentially similar to the process I initially described).

We might be okay to point to dev/gfdl to stay up to date.  However, since the NOAA-GFDL MOM6 maintainers won't be regression testing our use cases they won't know if/when a change breaks our models.  Do we worry about this with any of the other components?

A separate but related issue is that I don't think MOM6 is routinely tested with the latest dev/GFDL FMS version (mom6-examples points to release 2022.02).  SHiELD uses the latest FMS and updates w/ each release?

Comment (bensonr) [reply CHECKOUT_mom6.csh:13]: > A separate but related issue is that I don't think MOM6 is routinely tested with the latest dev/GFDL FMS version (mom6-examples points to release 2022.02). SHiELD uses the latest FMS and updates w/ each release?

It is because of this lack of MOM6 testing upstream dependencies, that the FMS dev team has integrated the MOM6 test suite into the automoated CI/CD system for FMS PRs.  See an example of it [here](https://github.com/NOAA-GFDL/FMS/actions/runs/10370070480/job/28707162037?pr=1572).

**Review by laurenchilutti — APPROVED** [review]

**Comment by breichl** [comment]
I chatted w/ Bob to get more insight into why the XMLs typically checkout MOM6 and SIS2 as submodules of MOM6-examples instead of directly checking out MOM6 and SIS2.  He said it is about ensuring MOM6 and SIS2 remain compatible (SIS2 relies on code in MOM6, so that certain versions of SIS2 will not work with certain versions of MOM6).  By using the submodules from MOM6-examples we only need to track one repository and are ensured the code bases are compatible and upgraded together.  So, a drawback of the approach proposed here is that we have some additional steps when we upgrade MOM6 and SIS2 so they are done consistently and not independently (this could be done by simply checking out the same versions in MOM6-examples dev/gfdl).

Regardless, I am okay with this change if it is preferred to avoid the additional overhead of MOM6-examples.  We just need to remain careful that updates to the MOM6 and SIS2 code bases are handled together.

**Pull request merged**
